packages/babel-parser/src/parser/expression.js

```diff
@@ -1425,108 +1430,119 @@ export default class ExpressionParser extends LValParser {
         if (this.eat(tt.braceR)) break;
       }
 
-      if (this.match(tt.at)) {
-        if (this.hasPlugin("decorators")) {
-          this.raise(
-            this.state.start,
-            "Stage 2 decorators disallow object literal property decorators",
-          );
-        } else {
-          // we needn't check if decorators (stage 0) plugin is enabled since it's checked by
-          // the call to this.parseDecorator
-          while (this.match(tt.at)) {
-            decorators.push(this.parseDecorator());
-          }
-        }
-      }
+      const prop = this.parseObjectMember(isPattern, refShorthandDefaultPos);
+      // $FlowIgnore RestElement will never be returned if !isPattern
+      if (!isPattern) this.checkPropClash(prop, propHash);
 
-      let prop = this.startNode(),
-        isGenerator = false,
-        isAsync = false,
-        startPos,
-        startLoc;
-      if (decorators.length) {
-        prop.decorators = decorators;
-        decorators = [];
-      }
-
-      if (this.match(tt.ellipsis)) {
-        prop = this.parseSpread(isPattern ? { start: 0 } : undefined);
-        node.properties.push(prop);
-        if (isPattern) {
-          this.toAssignable(prop, true, "object pattern");
-          this.checkCommaAfterRest(tt.braceR, "property");
-          this.expect(tt.braceR);
-          break;
-        }
-        continue;
-      }
-
-      prop.method = false;
-
-      if (isPattern || refShorthandDefaultPos) {
-        startPos = this.state.start;
-        startLoc = this.state.startLoc;
+      // $FlowIgnore
+      if (prop.shorthand) {
+        this.addExtra(prop, "shorthand", true);
       }
 
-      if (!isPattern) {
-        isGenerator = this.eat(tt.star);
-      }
+      node.properties.push(prop);
+    }
 
-      const containsEsc = this.state.containsEsc;
+    return this.finishNode(
+      node,
+      isPattern ? "ObjectPattern" : "ObjectExpression",
+    );
+  }
 
-      if (!isPattern && this.isContextual("async")) {
-        if (isGenerator) this.unexpected();
+  isAsyncProp(prop: N.ObjectProperty): boolean {
+    return (
+      !prop.computed &&
+      prop.key.type === "Identifier" &&
+      prop.key.name === "async" &&
+      (this.match(tt.name) ||
+        this.match(tt.num) ||
+        this.match(tt.string) ||
+        this.match(tt.bracketL) ||
+        this.state.type.keyword ||
+        this.match(tt.star)) &&
+      !lineBreak.test(
+        this.state.input.slice(this.state.lastTokEnd, this.state.start),
```

Comment:
You can use `this.hasPrecedingLineBreak`

Reply:
Done